ui/src/utils/fetchWrappers/pipelineFetch.test.ts

```diff
@@ -0,0 +1,80 @@
+// Tests pipelineFetch.ts
+
+import { usePipelineSummaryFetch } from "./pipelineFetch";
+import { useFetch } from "./fetch";
+import { renderHook } from "@testing-library/react";
+
+jest.mock("../fetchWrappers/fetch");
+const mockedUseFetch = useFetch as jest.MockedFunction<typeof useFetch>;
+
+describe("pipelineFetch test", () => {
+  const mockData = {
+    data: {
+      id: "test-pipeline",
+      name: "test-pipeline",
+      namespace: "default",
+      status: "Running",
+      isb: "test-isb",
+      isbStatus: "Running",
+      isbNamespace: "default",
+      isbService: "test-isb",
+      isbServicePort: 8080,
+      isbServicePath: "/",
+      isbServiceUrl: "http://test-isb.default.svc.cluster.local:8080/",
+      isbServiceType: "ClusterIP",
+      isbServiceClusterIP: "",
+      isbServiceExternalIPs: [],
+    },
+  };
+
+  it("should return pipeline data", async () => {
+    mockedUseFetch.mockImplementation(() => ({
+      data: mockData,
+      error: null,
+      loading: false,
+    }));
+
+    const { result } = renderHook(() =>
+      usePipelineSummaryFetch({
+        namespaceId: "default",
+        pipelineId: "test-pipeline",
+        addError: null,
+      })
+    );
+    expect(result?.current?.data?.pipelineData.name).toEqual("test-pipeline");
+  });
+
+  it("should return undefined pipeline data when loading", async () => {
+    mockedUseFetch.mockImplementation(() => ({
+      data: null,
+      error: null,
+      loading: true,
+    }));
+
+    const { result } = renderHook(() =>
+      usePipelineSummaryFetch({
+        namespaceId: "default",
+        pipelineId: "test-pipeline",
+        addError: null,
+      })
+    );
+    expect(result?.current?.data?.pipelineData).toEqual(undefined);
+  });
+
+  it("should return error", async () => {
+    mockedUseFetch.mockImplementation(() => ({
+      data: null,
+      error: "error",
+      loading: false,
+    }));
+
+    const { result } = renderHook(() =>
+      usePipelineSummaryFetch({
+        namespaceId: "default",
+        pipelineId: "test-pipeline",
+        addError: null,
+      })
+    );
+    expect(result?.current?.error).toEqual("error");
+  });
+});
```

Comment:
can we add test cases for `isb-services` fetch call as well

Reply:
Done!